rs/messaging/src/routing/stream_handler/tests.rs

```diff
@@ -83,520 +79,411 @@ fn oversized_reject_message_is_truncated() {
+    induct_loopback_stream_ignores_memory_limit_impl(HypervisorConfig {
+        subnet_memory_capacity: NumBytes::new(MAX_RESPONSE_COUNT_BYTES as u64 * 7 / 2),
+        ..Default::default()
     });
 }
 
 /// Tests that subnet message memory limit is ignored by
 /// `StreamHandlerImpl::induct_loopback_stream()` for system subnets.
 #[test]
 fn system_subnet_induct_loopback_stream_ignores_subnet_message_memory_limit() {
-    with_test_replica_logger(|log| {
-        // A stream handler with a subnet message memory limit that only allows up to 3 reservations.
-        let config = HypervisorConfig {
-            subnet_message_memory_capacity: NumBytes::new(MAX_RESPONSE_COUNT_BYTES as u64 * 7 / 2),
-            ..Default::default()
-        };
-        let (stream_handler, mut initial_state, metrics_registry) =
-            new_fixture_with_config(&log, config);
-        initial_state.metadata.own_subnet_type = SubnetType::System;
-
-        induct_loopback_stream_ignores_memory_limit_impl(
-            stream_handler,
-            initial_state,
-            metrics_registry,
-        );
+    // A stream handler with a subnet message memory limit that only allows up to 3 reservations.
+    induct_loopback_stream_ignores_memory_limit_impl(HypervisorConfig {
+        subnet_message_memory_capacity: NumBytes::new(MAX_RESPONSE_COUNT_BYTES as u64 * 7 / 2),
+        ..Default::default()
     });
 }
 
 /// Tests that subnet wasm custom sections memory limit is ignored by
 /// `StreamHandlerImpl::induct_loopback_stream()` for system subnets.
 #[test]
 fn system_subnet_induct_loopback_stream_ignores_subnet_wasm_custom_sections_memory_limit() {
-    with_test_replica_logger(|log| {
-        // A stream handler with a subnet message memory limit that only allows up to 3 reservations.
-        let config = HypervisorConfig {
-            subnet_message_memory_capacity: NumBytes::new(MAX_RESPONSE_COUNT_BYTES as u64 * 7 / 2),
-            subnet_wasm_custom_sections_memory_capacity: NumBytes::new(0),
-            ..Default::default()
-        };
-        let (stream_handler, mut initial_state, metrics_registry) =
-            new_fixture_with_config(&log, config);
-        initial_state.metadata.own_subnet_type = SubnetType::System;
-
-        induct_loopback_stream_ignores_memory_limit_impl(
-            stream_handler,
-            initial_state,
-            metrics_registry,
-        );
+    // A stream handler with a subnet message memory limit that only allows up to 3 reservations.
+    induct_loopback_stream_ignores_memory_limit_impl(HypervisorConfig {
+        subnet_message_memory_capacity: NumBytes::new(MAX_RESPONSE_COUNT_BYTES as u64 * 7 / 2),
+        subnet_wasm_custom_sections_memory_capacity: NumBytes::new(0),
+        ..Default::default()
     });
 }
 
+/// Common initial state setup for `StreamHandlerImpl::induct_loopback_stream()`
+/// memory limit tests.
+fn with_induct_loopback_stream_with_memory_limit_setup(
```

Comment:
Nit: This is a very convoluted name. I'd at least drop the `_with_memory_limit` part, since that comes (or not) from the `HypervisorConfig` argument.

Reply:
Renamed.